src/SFML/Network/IpAddress.cpp

```diff
@@ -36,7 +36,7 @@ namespace
     sfIpAddress fromSFMLAddress(sf::IpAddress address)
     {
         sfIpAddress result;
-        strncpy(result.address, address.toString().c_str(), 16);
+        strncpy(result.address, address.toString().c_str(), 15);
```

Comment:
No `\0` terminator?

Reply:
`result.address` is a `char` array of length 16. That means it can only fit 15 character + `\0`. If you call [`strncpy`](https://en.cppreference.com/w/cpp/string/byte/strncpy) with a count of 16 and the string you're copying from with 16 characters or longer, you end up copying the first 16 non-null character into that array of 16 characters and are left with a string that is not null terminated. If the string is not null terminated, you're prone to accidentally read off the end of that array.

Reply:
Ah, I see

Reply:
Don't you need to set the last char? I believe it is not initialized.
`result.address[15] = '\0';`

Reply:
`strncpy` fills in all remaining characters will null so we don’t have to do that manually

Reply:
Good point. We may want to initialize that buffer to all zeros first.

Reply:
https://github.com/SFML/CSFML/compare/515ef0ef193f3f8332be271f0aca186316d68d3c..5bdfdb4b10d25134ad2d8b877ef519958e3f2a30

I made sure that we're initializing the `sfIpAddress` before writing to it. I wish there was a more elegant solution but this is the best I could come up with.